core/iwasm/common/wasm_memory.c

```diff
@@ -147,23 +147,70 @@ wasm_runtime_free_internal(void *ptr)
 void *
 wasm_runtime_malloc(unsigned int size)
 {
+#if 0
     if (size == 0) {
         LOG_WARNING("warning: wasm_runtime_malloc with size zero\n");
         /* At lease alloc 1 byte to avoid malloc failed */
         size = 1;
     }
 
     return wasm_runtime_malloc_internal(size);
+#endif
+    return wasm_gc_malloc(size);
 }
 
 void *
 wasm_runtime_realloc(void *ptr, unsigned int size)
 {
+#if 0
     return wasm_runtime_realloc_internal(ptr, size);
+#endif
+    return NULL;
 }
 
 void
 wasm_runtime_free(void *ptr)
+{
+#if 0
+  wasm_runtime_free_internal(ptr);
+#endif
+  wasm_gc_free(ptr);
+}
+
+#if WASM_ENABLE_GC != 0
+static inline void *
+wasm_gc_malloc_internal(unsigned int size)
+{
+    return mem_allocator_malloc_with_gc(pool_allocator, size);
+}
+ 
+void *
+wasm_gc_malloc(unsigned int size)
```

Comment:
Could we change the function prototype to `void *wasm_runtime_gc_malloc(void *heap_handle, unsigned int size)`, my understanding is that we should not reuse the global heap's pool allocator, but we should initialize another global gc heap (as the global heap might be not created if developer initializes runtime with other allocation mode) or initialize gc heap for each module instance.

Reply:
And like wasm_module_malloc, we can directly call `mem_allocator_malloc_with_gc(heap_handle, size);`